src/middleware/AuthMiddleware.js

```diff
@@ -1,179 +1,187 @@
   };
 
   const { error } = Joi.validate(req.body, userSchema);
   if (error) {
-    return res.status(400).send(error.details);
+    return res.status(400).json({ message: error.details[0].message });
   }
 
   const { password } = req.body;
   req.body.password = await bcrypt.hash(password, 10);
   return next();
 };
 
+/**
+ * @description - this method checks for phone and email in the database. if exist
+ * it return object message, otherwise it call the next middleware
+ *
+ * @param {object} req - request
+ *
+ * @param {object} res - response
+ *
+ * @param {object} next - it call the next middleware in the stack
+ *
+ * @returns {object} - it return object
+ */
 export const checkUserDetails = async (req, res, next) => {
-  // Check if email and phone exist.
   const { email, phone } = req.body;
   try {
     const emailExists = await userModel.select('*', `WHERE "email" = '${email}'`);
     const phoneNumberExists = await userModel.select('*', `WHERE "phone" = '${phone}'`);
     if (emailExists.rowCount) {
-      return res.status(400).send({
+      return res.status(404).send({
         message: 'Email already exists',
         status: false
       });
     }
 
     if (phoneNumberExists.rowCount) {
-      return res.status(400).send({
+      return res.status(404).send({
         message: 'Phone number already exists',
         status: false
       });
     }
     return next();
-  } catch (error) {
-    res.send({ erroe: `${error.message}` });
+  } catch (err) {
+    res.send({ error: `${err.message}` });
   }
 };
 
+/**
+ *
+ * @description - this method checks user loggin details and return
+ * token and user objects
+ *
+ * @param {object} req - request
+ *
+ * @param {object} res - response
+ *
+ * @returns {object} - it return data object
+ */
 export const loginUser = async (req, res) => {
   const { password, email } = req.body;
   try {
-    // Select user with req.body.email, if not exist send error
     const user = await userModel.select('*', `WHERE "email" = '${email}'`);
     if (!user.rowCount) {
-      return res.status(400).send({ message: 'Email does not exist' });
+      return res.status(404).send({ message: 'Email does not exist' });
     }
-    // compare crypted password and see if it match, if not match send error
     const passwordIsValid = await bcrypt.compare(password, user.rows[0].password);
     if (!passwordIsValid) {
-      res.status(400).send({ message: 'Password does correct' });
+      res.status(404).send({ message: 'Password does correct' });
     }
-    // if password is valid
-    const { id, first_name } = user.rows[0];
+    const { id, firstName } = user.rows[0];
     const userData = {
       id,
-      first_name
+      firstName
     };
-    const token = jwt.sign({
-      userData
-    }, secretKey, {
-      expiresIn: '24h'
-    });
-    return res.status(200).json({
-      msg: 'Welcome',
+    const token = assignToken(userData);
+    return res.status(201).json({
```

Comment:
When an operation is successful without a "write to the database" then `200` (OK) status code is ideal instead of `201` (CREATED)